public/app/plugins/datasource/loki/language_provider.ts

```diff
@@ -269,7 +269,7 @@ export default class LokiLanguageProvider extends LanguageProvider {
       selector = EMPTY_SELECTOR;
     }
 
-    if (!isValueStart && selector === EMPTY_SELECTOR) {
+    if (!labelKey && !isValueStart && selector === EMPTY_SELECTOR) {
```

Comment:
Could it be simplified to: `!labelKey && selector === EMPTY_SELECTOR`?

 It seems to be covering the case when the very first label is being typed. It can only happen when the selector is empty and no label is fully defined yet. Once the label is defined the value of `isValueStart` won't matter  so it could be removed from the condition.

Reply:
Yes, you are correct. Nice catch.

Reply:
Changed in 3e204e2